common/lib/exit.c

```diff
@@ -0,0 +1,13 @@
+#include <stdlib.h>
+#include <stdio.h>
+
+extern void __exit(int status);
+
+void exit(int status) {
+  __exit(status);
+
+  // Prevent the compiler from considering this entire basic block unreachable.
+  asm volatile ("");
```

Comment:
Instead, just mark __exit as attribute((noreturn)). This is done automatically for abort() and exit()